packages/nightly-connect/src/lib/nightly-connect.ts

```diff
@@ -0,0 +1,223 @@
+    [];
+  let client: AppNear | undefined;
+  const modal = new NightlyConnectModal();
+
+  const getAccounts = () => {
+    return connectedAccounts;
+  };
+  const signOut = async (): Promise<void> =>
+    new Promise<void>((resolve, reject) => {
+      const timer = setTimeout(() => {
+        reject("Closing connection timed out");
+      }, 5000);
+      client!.ws.onclose = () => {
+        clearTimeout(timer);
+        client = undefined;
+        connectedAccounts = [];
+        resolve();
+      };
+      client?.ws.close();
+    });
+
+  const transformTransactions = (
+    transactions: Array<Optional<Transaction, "signerId" | "receiverId">>
+  ): Array<Transaction> => {
+    const accounts = getAccounts();
+    const { contract } = store.getState();
+
+    if (!accounts.length || !contract) {
+      throw new Error("Wallet not signed in");
+    }
+
+    return transactions.map((transaction) => {
+      return {
+        signerId: transaction.signerId || accounts[0].accountId,
+        receiverId: transaction.receiverId || contract.contractId,
+        actions: transaction.actions,
+      };
+    });
+  };
+  const signer: Signer = {
+    createKey: () => {
+      throw new Error("Not implemented");
+    },
+    getPublicKey: async (accountId) => {
+      const accounts = getAccounts();
+      const account = accounts.find((a) => a.accountId === accountId);
+
+      if (!account) {
+        throw new Error("Failed to find public key for account");
+      }
+
+      return utils.PublicKey.from(account.publicKey);
+    },
+    signMessage: async (message, accountId) => {
+      const accounts = getAccounts();
+      const account = accounts.find((a) => a.accountId === accountId);
+
+      if (!account) {
+        throw new Error("Failed to find account for signing");
+      }
+
+      try {
+        const tx = nearTransactions.Transaction.decode(Buffer.from(message));
+        const signedTx = await client!.signTransaction(tx);
+
+        return {
+          signature: signedTx.signature.data,
+          publicKey: tx.publicKey,
+        };
+      } catch (err) {
+        logger.log("Failed to sign message");
+        logger.error(err);
+
+        throw Error("Invalid message. Only transactions can be signed");
+      }
+    },
+  };
+  return {
+    async signIn() {
+      return new Promise((resolve, reject) => {
+        const existingAccounts = getAccounts();
+
+        if (existingAccounts.length) {
+          return resolve(existingAccounts);
+        }
+
+        try {
+          setupNightlyConnectState({
+            ...params,
+            icon: params.appIcon,
+            additionalInfo: params.additionalInfo || "",
+            onUserConnect: (pk) => {
+              connectedAccounts.push(pk);
+              modal.closeModal();
+              resolve(getAccounts());
+            },
+          }).then((appClient) => {
+            client = appClient;
+            modal.openModal(client.sessionId, Network.Near);
```

Comment:
Does `Network` clash with `options.network` here? How does Nightly Connect handle the different networks in NEAR or is that not concern?

Reply:
Overall it does not matter if you are signing on Devnet or Mainnet signature should be valid on both. `Nightly newtork` is referring to Solana/Near etc.